models/Monte_Carlo.py

```diff
@@ -0,0 +1,166 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from arc import *
+import scipy.constants as const
+from scipy.integrate import odeint
+
+def Monte_Carlo_iHO_3d(atom, t, omega_H, omega_L, T, N_ensemble, traping_laser, D2_line):
+    
+    def wavelength2frequency (wavelength):
+        from scipy.constants import c as c_c
+        return c_c/(wavelength)
+    
+    def iHO_a_3d (x, y, z, omega_H, omega_L):
+        ax = (omega_H**2)*x
+        ay = (omega_H**2)*y
+        g = 9.8 #m/s^2 gravity
+        az = (omega_L**2)*z - g
+        return ax, ay, az
+    
+    def newton_3d(f, t, omega_H, omega_L):
+        x, vx, y, vy, z, vz = f
+        ax, ay, az = iHO_a_3d(x, y, z, omega_H, omega_L)
+        dfdt = [vx, ax, vy, ay, vz, az]
+        return dfdt
+
+    m = atom.mass
+    delta_x = np.sqrt(const.k*T/(m*omega_H**2))
+    delta_vx = np.sqrt(const.k*T/m)
+    delta_y = np.sqrt(const.k*T/(m*omega_H**2))
+    delta_vy = np.sqrt(const.k*T/m)
+    delta_z = np.sqrt(const.k*T/(m*omega_L**2))
+    delta_vz = np.sqrt(const.k*T/m)
+    x_ensemble = np.random.normal(0, delta_x, N_ensemble)
+    y_ensemble = np.random.normal(0, delta_y, N_ensemble)
+    z_ensemble = np.random.normal(0, delta_z, N_ensemble)
+    vx_ensemble = np.random.normal(0, delta_vx, N_ensemble)
+    vy_ensemble = np.random.normal(0, delta_vy, N_ensemble)
+    vz_ensemble = np.random.normal(0, delta_vz, N_ensemble)
+
```

Comment:
!ETU !RFC: As an example of vectorization, this entire block could be condensed into four lines expressing your intent more clearly:

```python
delta_r = np.sqrt(const.k * T / (m * np.array([omega_H, omega_H, omega_L])**2))
delta_v = np.sqrt(const.k * T / m) * np.ones(3)
r_ensemble = np.random.normal(size=(N_ensemble, 3)) * delta_r
v_ensemble = np.random.normal(size=(N_ensemble, 3)) * delta_v
```

Whenever variables broken up into $x,y,z$ components here, it should be possible to rewrite the code more briefly and clearly in vector form.